plugins/plugin-java/che-plugin-java-plain/che-plugin-java-plain-server/src/main/java/org/eclipse/che/plugin/java/plain/server/generator/PlainJavaProjectGenerator.java

```diff
@@ -74,7 +90,51 @@ public void onCreateProject(
         mainJavaWsPath,
         getClass().getClassLoader().getResourceAsStream("files/main_class_content"));
 
-    service.createSimpleProject(prefixURI(projectWsPath), sourceFolders.get(0));
+    createClasspath(projectWsPath, sourceFolders.get(0));
+    createProjectConfig(projectWsPath);
+  }
+
+  private void createProjectConfig(String projectWsPath)
+      throws ConflictException, NotFoundException, ServerException {
+    InputStream projectIS =
+        getClass().getClassLoader().getResourceAsStream("files/project_config_content");
+    try {
+
+      String projectConfigTemplate = IOUtils.toString(projectIS);
+      String projectConfWsPath = resolve(projectWsPath, ".project");
+      String projectConfContent =
+          projectConfigTemplate.replaceFirst("project_name", projectWsPath.substring(1));
```

Comment:
Create constants for all template names.

Reply:
OK